### ISSUE TITLE
bgp-announce-lb-ip with kube-proxy-replacement=probe, traffic fails with load balancer ip

### ISSUE THREAD

**Issue by tedli** [cilium/cilium#23010]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I have a working cluster. I want to evaluate the `loadbalancer` feature. BGP announced with no error reported, ingress ip assigned in service status.

But access that load balancer ip outside cluster fails, meanwhile inside cluster this ip works.

If I config `kube-proxy-replacement` to `disabled`, load balancer ip works both inside and outside cluster.

### Cilium Version

v1.11.2

### Kernel Version

4.19.91-24.8.an8.x86_64

### Kubernetes Version

v1.19.7

### Relevant log output

My cilium config map:

```yaml
apiVersion: v1
data:
  auto-direct-node-routes: "true"
  bgp-announce-lb-ip: "true"
  bpf-lb-acceleration: native
  bpf-lb-algorithm: maglev
  bpf-lb-mode: dsr
  bpf-map-dynamic-size-ratio: "0.0025"
  bpf-policy-map-max: "16384"
  cluster-name: default
  cluster-pool-ipv4-cidr: 172.24.68.0/22
  cluster-pool-ipv4-mask-size: "24"
  debug: "false"
  disable-cnp-status-updates: "true"
  enable-auto-protect-node-port-range: "true"
  enable-bpf-clock-probe: "true"
  enable-bpf-masquerade: "false"
  enable-endpoint-health-checking: "true"
  enable-ipv4: "true"
  enable-ipv4-masquerade: "false"
  enable-ipv6: "false"
  enable-remote-node-identity: "true"
  enable-session-affinity: "true"
  enable-well-known-identities: "false"
  enable-xt-socket-fallback: "true"
  identity-allocation-mode: crd
  install-iptables-rules: "true"
  ipam: cluster-pool
  k8s-require-ipv4-pod-cidr: "true"
  k8s-require-ipv6-pod-cidr: "false"
  kube-proxy-replacement: probe
  monitor-aggregation: medium
  monitor-aggregation-flags: all
  monitor-aggregation-interval: 5s
  node-port-bind-protection: "true"
  node-port-mode: dsr
  operator-api-serve-addr: 127.0.0.1:9234
  preallocate-bpf-maps: "false"
  sidecar-istio-proxy-image: cilium/istio_proxy
  tofqdns-enable-poller: "false"
  tunnel: disabled
  wait-bpf-mount: "false"
```

cilium status output:
```shell
KVStore:                Ok   Disabled
Kubernetes:             Ok   1.19 (v1.19.7) [linux/amd64]
Kubernetes APIs:        ["cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "discovery/v1beta1::EndpointSlice", "networking.k8s.io/v1::NetworkPolicy"]
KubeProxyReplacement:   Probe   [eth2 10.206.67.114 (Direct Routing)]
Host firewall:          Disabled
Cilium:                 Ok   1.11.2 (v1.11.2-17b0dbd)
NodeMonitor:            Disabled
Cilium health daemon:   Ok   
IPAM:                   IPv4: 11/254 allocated from 172.24.68.0/24, 
Allocated addresses:
  172.24.68.119 (kstone/kstone-inspection-controller-fcb4fb4fd-jwbdk[restored])
  172.24.68.123 (kube-system/kstone-tapp-69b7997cf5-x5b8t[restored])
  172.24.68.130 (kstone/kstone-dashboard-api-569978ff7f-gkzq9[restored])
  172.24.68.133 (router)
  172.24.68.154 (kstone/kstone-etcd-controller-7c848f6bff-qj7qb[restored])
  172.24.68.211 (kstone/kstone-grafana-7764d7dcdb-lrjmp[restored])
  172.24.68.222 (kstone/test12-etcd-2[restored])
  172.24.68.245 (health)
  172.24.68.253 (lm-test/kubecluster-sample-1-21-3-etcd-1[restored])
  172.24.68.4 (kube-system/open-local-controller-58dfb5fff-gt6qj[restored])
  172.24.68.59 (qihoo-system/loadbalancers-babel-controller-manager-55665c89f6-6hvz9[restored])
BandwidthManager:       Disabled
Host Routing:           Legacy
Masquerading:           Disabled
Clock Source for BPF:   ktime
Controller Status:      66/66 healthy
  Name                                          Last success   Last error   Count   Message
  bpf-map-sync-cilium_lxc                       5s ago         never        0       no error   
  cilium-health-ep                              59s ago        never        0       no error   
  dns-garbage-collector-job                     7s ago         never        0       no error   
  endpoint-1060-regeneration-recovery           never          never        0       no error   
  endpoint-1628-regeneration-recovery           never          never        0       no error   
  endpoint-193-regeneration-recovery            never          never        0       no error   
  endpoint-197-regeneration-recovery            never          never        0       no error   
  endpoint-271-regeneration-recovery            never          never        0       no error   
  endpoint-2844-regeneration-recovery           never          never        0       no error   
  endpoint-2850-regeneration-recovery           never          never        0       no error   
  endpoint-441-regeneration-recovery            never          never        0       no error   
  endpoint-54-regeneration-recovery             never          never        0       no error   
  endpoint-55-regeneration-recovery             never          never        0       no error   
  endpoint-826-regeneration-recovery            never          never        0       no error   
  endpoint-gc                                   4m7s ago       never        0       no error   
  ipcache-inject-labels                         18m31s ago     34m6s ago    0       no error   
  k8s-heartbeat                                 7s ago         never        0       no error   
  mark-k8s-node-as-available                    34m0s ago      never        0       no error   
  metricsmap-bpf-prom-sync                      2s ago         never        0       no error   
  neighbor-table-refresh                        0s ago         never        0       no error   
  resolve-identity-271                          3m59s ago      never        0       no error   
  restoring-ep-identity (1060)                  33m35s ago     never        0       no error   
  restoring-ep-identity (1628)                  33m35s ago     never        0       no error   
  restoring-ep-identity (193)                   33m35s ago     never        0       no error   
  restoring-ep-identity (197)                   33m35s ago     never        0       no error   
  restoring-ep-identity (2844)                  34m0s ago      never        0       no error   
  restoring-ep-identity (2850)                  33m35s ago     never        0       no error   
  restoring-ep-identity (441)                   33m35s ago     never        0       no error   
  restoring-ep-identity (54)                    33m35s ago     never        0       no error   
  restoring-ep-identity (55)                    33m35s ago     never        0       no error   
  restoring-ep-identity (826)                   33m35s ago     never        0       no error   
  sync-endpoints-and-host-ips                   0s ago         never        0       no error   
  sync-lb-maps-with-k8s-services                34m0s ago      never        0       no error   
  sync-policymap-1060                           30s ago        never        0       no error   
  sync-policymap-1628                           30s ago        never        0       no error   
  sync-policymap-193                            30s ago        never        0       no error   
  sync-policymap-197                            30s ago        never        0       no error   
  sync-policymap-271                            30s ago        never        0       no error   
  sync-policymap-2844                           30s ago        never        0       no error   
  sync-policymap-2850                           30s ago        never        0       no error   
  sync-policymap-441                            30s ago        never        0       no error   
  sync-policymap-54                             30s ago        never        0       no error   
  sync-policymap-55                             30s ago        never        0       no error   
  sync-policymap-826                            30s ago        never        0       no error   
  sync-to-k8s-ciliumendpoint (1060)             5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (1628)             5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (193)              5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (197)              5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (271)              9s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (2844)             0s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (2850)             5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (441)              5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (54)               5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (55)               5s ago         never        0       no error   
  sync-to-k8s-ciliumendpoint (826)              5s ago         never        0       no error   
  template-dir-watcher                          never          never        0       no error   
  update-k8s-node-annotations                   34m6s ago      never        0       no error   
  waiting-initial-global-identities-ep (1060)   33m35s ago     never        0       no error   
  waiting-initial-global-identities-ep (1628)   33m35s ago     never        0       no error   
  waiting-initial-global-identities-ep (193)    33m35s ago     never        0       no error   
  waiting-initial-global-identities-ep (197)    33m35s ago     never        0       no error   
  waiting-initial-global-identities-ep (2850)   33m35s ago     never        0       no error   
  waiting-initial-global-identities-ep (441)    33m35s ago     never        0       no error   
  waiting-initial-global-identities-ep (54)     33m35s ago     never        0       no error   
  waiting-initial-global-identities-ep (55)     33m35s ago     never        0       no error   
  waiting-initial-global-identities-ep (826)    33m35s ago     never        0       no error   
Proxy Status:   OK, ip 172.24.68.133, 0 redirects active on ports 10000-20000
Hubble:         Disabled
KubeProxyReplacement Details:
  Status:                 Probe
  Socket LB Protocols:    TCP, UDP
  Devices:                eth2 10.206.67.114 (Direct Routing)
  Mode:                   DSR
  Backend Selection:      Maglev (Table Size: 16381)
  Session Affinity:       Enabled
  Graceful Termination:   Enabled
  XDP Acceleration:       Native
  Services:
  - ClusterIP:      Enabled
  - NodePort:       Enabled (Range: 30000-32767) 
  - LoadBalancer:   Enabled 
  - externalIPs:    Enabled 
  - HostPort:       Enabled
BPF Maps:   dynamic sizing: on (ratio: 0.002500)
  Name                          Size
  Non-TCP connection tracking   589737
  TCP connection tracking       1179474
  Endpoint policy               65535
  Events                        40
  IP cache                      512000
  IP masquerading agent         16384
  IPv4 fragmentation            8192
  IPv4 service                  65536
  IPv6 service                  65536
  IPv4 service backend          65536
  IPv6 service backend          65536
  IPv4 service reverse NAT      65536
  IPv6 service reverse NAT      65536
  Metrics                       1024
  NAT                           1179474
  Neighbor table                1179474
  Global policy                 16384
  Per endpoint policy           65536
  Session affinity              65536
  Signal                        40
  Sockmap                       65535
  Sock reverse NAT              589737
  Tunnel                        65536
Encryption:                                         Disabled
Cluster health:                                     3/3 reachable   (2023-01-10T09:26:41Z)
  Name                                              IP              Node        Endpoints
  docker13.xxxx (localhost)   10.206.67.114   reachable   reachable
  docker11.xxxx               10.206.67.95    reachable   reachable
  docker12.xxxx               10.206.67.96    reachable   reachable
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tedli** [comment]
I found this issue https://github.com/cilium/cilium/issues/11724, I think my case is different.

**Comment by tedli** [comment]
I tried with v1.12.5, still failed if `kube-proxy-replacement` not `disabled`.

```bash
helm install cilium cilium/cilium --version 1.12.5 --namespace kube-system --set bgp.enabled=true --set bgp.announce.loadbalancerIP=true --set image.repository=xxxxx/cilium/cilium --set hubble.enabled=false --set ipam.operator.clusterPoolIPv4PodCIDR=172.24.68.0/22 --set tunnel=disabled --set operator.image.repository=xxxxx/cilium/operator --set operator.replicas=1 --set autoDirectNodeRoutes=true --set ipv4NativeRoutingCIDR=172.24.68.0/22 --set loadBalancer.algorithm=maglev --set loadBalancer.mode=dsr --set loadBalancer.acceleration=native
```

**Comment by tedli** [comment]
My cluster was setup using rancher1, and there are 3 nodes. `kube proxy replacement` works as expect with `disabled`, `probe` and `strick` for service cluster ip, however the bgp based lb ip works only with `disabled`.
Meantime, there was also a weird thing, if I schedule the workload pod (nginx) to node `docker11.xxxxxxxxxx`, the lb ip works as expect with `probe` or `strick`. If schedule pod to other two node, traffic hang (*NOT* not route to host, connect refuse, reset by peer, *BUT* timeout).

```
[xxxxxxx@docker11 ~]$ kubectl get no -o wide
NAME                                  STATUS   ROLES                      AGE   VERSION   INTERNAL-IP     EXTERNAL-IP   OS-IMAGE        KERNEL-VERSION            CONTAINER-RUNTIME
docker11.xxxxxxxxxx   Ready    controlplane,etcd,worker   84d   v1.19.7   10.206.67.95    <none>        Anolis OS 8.2   4.19.91-24.8.an8.x86_64   docker://20.10.18
docker12.xxxxxxxxxx   Ready    controlplane,etcd,worker   86d   v1.19.7   10.206.67.96    <none>        Anolis OS 8.2   4.19.91-24.8.an8.x86_64   docker://20.10.18
docker13.xxxxxxxxxx   Ready    controlplane,etcd,worker   86d   v1.19.7   10.206.67.114   <none>        Anolis OS 8.2   4.19.91-24.8.an8.x86_64   docker://20.10.18
```

**Comment by tedli** [comment]
On `strict` mode (so did `probe`)

```bash
[xxxxxx@docker11 ~]$ kubectl get svc
NAME              TYPE           CLUSTER-IP        EXTERNAL-IP     PORT(S)        AGE
kubernetes        ClusterIP      192.168.0.1       <none>          443/TCP        86d
nginx             ClusterIP      192.168.160.55    <none>          80/TCP         15d
nginx-lb-fixed    LoadBalancer   192.168.135.203   10.218.112.88   80:30415/TCP   8d
nginx-lvm         ClusterIP      None              <none>          80/TCP         86d
nginx1            ClusterIP      192.168.146.202   <none>          80/TCP         15d
nginx1-lb-fixed   LoadBalancer   192.168.57.128    10.218.112.99   80:31413/TCP   8d
[xxxxxx@docker11 ~]$ kubectl get po -o wide
NAME                      READY   STATUS    RESTARTS   AGE    IP              NODE                                  NOMINATED NODE   READINESS GATES
nginx-55ff6479f9-td8vc    1/1     Running   0          151m   172.24.68.121   docker13.xxxxxxxxxx   <none>           <none>
nginx-lvm-0               1/1     Running   4          9d     172.24.70.35    docker12.xxxxxxxxxx   <none>           <none>
nginx1-5cd95d88d7-ncnzl   1/1     Running   0          8d     172.24.69.237   docker11.xxxxxxxxxx   <none>           <none>
```

There are `88` and `99`, which have one nginx pod respectively.

```bash
[xxxxxxx@docker11 ~]$ kubectl get po -n kube-system -l k8s-app=cilium -o wide
NAME           READY   STATUS    RESTARTS   AGE    IP              NODE                                  NOMINATED NODE   READINESS GATES
cilium-7q56z   1/1     Running   0          158m   10.206.67.96    docker12.xxxxxxxxxx   <none>           <none>
cilium-pt4pf   1/1     Running   0          159m   10.206.67.114   docker13.xxxxxxxxxx   <none>           <none>
cilium-qqmkn   1/1     Running   0          159m   10.206.67.95    docker11.xxxxxxxxxx   <none>           <none>
```
I did getting map on `docker11`, there was the `99` entry, and `99` indeed works as expect.
```bash
[xxxxxxx@docker11 ~]$ kubectl exec -ti cilium-qqmkn -c cilium-agent -n kube-system -- cilium map get cilium_lb4_services_v2
Key                    Value                 State   Error
192.168.150.158:443    0 1 (9) [0x0 0x0]     sync    
192.168.115.124:2379   0 1 (15) [0x0 0x0]    sync    
192.168.57.128:80      0 1 (53) [0x0 0x0]    sync    
192.168.141.18:2379    0 1 (20) [0x0 0x0]    sync    
192.168.211.145:80     0 1 (31) [0x0 0x0]    sync    
10.206.67.95:32068     0 1 (72) [0x42 0x0]   sync    
0.0.0.0:32068          0 1 (73) [0x2 0x0]    sync    
192.168.0.2:53         0 2 (29) [0x0 0x0]    sync    
10.206.67.95:31413     0 1 (74) [0x42 0x0]   sync    
0.0.0.0:31882          0 1 (63) [0x2 0x0]    sync    
10.206.67.95:32351     0 1 (58) [0x42 0x0]   sync    
0.0.0.0:30708          0 1 (65) [0x2 0x0]    sync    
192.168.76.253:80      0 1 (4) [0x0 0x0]     sync    
192.168.131.64:443     0 1 (22) [0x0 0x0]    sync    
0.0.0.0:31413          0 1 (75) [0x2 0x0]    sync    
192.168.130.67:80      0 1 (18) [0x0 0x0]    sync    
10.206.67.95:30373     0 1 (66) [0x42 0x0]   sync    
192.168.114.35:443     0 1 (16) [0x0 0x0]    sync    
0.0.0.0:30403          0 1 (71) [0x2 0x0]    sync    
192.168.191.251:8443   0 0 (28) [0x0 0x0]    sync    
0.0.0.0:30738          0 1 (69) [0x2 0x0]    sync    
192.168.136.55:443     0 1 (13) [0x0 0x0]    sync    
192.168.205.177:443    0 3 (32) [0x0 0x0]    sync    
10.206.67.95:31882     0 1 (62) [0x42 0x0]   sync    
10.206.67.95:30708     0 1 (64) [0x42 0x0]   sync    
192.168.86.153:53      0 2 (5) [0x0 0x0]     sync    
192.168.146.202:80     0 1 (17) [0x0 0x0]    sync    
192.168.161.81:443     0 1 (21) [0x0 0x0]    sync    
192.168.76.192:23000   0 1 (10) [0x0 0x0]    sync    
192.168.25.58:9090     0 1 (27) [0x0 0x0]    sync    
192.168.99.185:9402    0 1 (14) [0x0 0x0]    sync    
10.218.112.99:80       0 1 (54) [0x60 0x0]   sync    <----- here
0.0.0.0:32351          0 1 (59) [0x2 0x0]    sync    
0.0.0.0:30373          0 1 (67) [0x2 0x0]    sync    
192.168.149.82:443     0 1 (26) [0x0 0x0]    sync    
10.206.67.95:30738     0 1 (68) [0x42 0x0]   sync    
10.206.67.95:30403     0 1 (70) [0x42 0x0]   sync    
192.168.0.1:443        0 3 (23) [0x0 0x0]    sync    
192.168.0.2:9153       0 2 (30) [0x0 0x0]    sync    
192.168.14.41:80       0 1 (19) [0x0 0x0]    sync 
```
However, doing the same, getting map on `docker12`, `docker13`, there was no `88` entry, nor did on `docker11`. And `88` indeed didn't work, traffic fails.
